servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -138,6 +140,10 @@ final class DefaultSingleAddressHttpClientBuilder<U, R> implements SingleAddress
             final U address, final ServiceDiscoverer<U, R, ? extends ServiceDiscovererEvent<R>> serviceDiscoverer) {
         this.address = requireNonNull(address);
         config = new HttpClientConfig();
+        // Skip default options not supported by this address type
+        if (!(address instanceof DomainSocketAddress || address instanceof io.netty.channel.unix.DomainSocketAddress)) {
```

Comment:
Another discovery is that even if users use `io.servicetalk.transport.api.DomainSocketAddress` to start a server, the `HttpServerContext.localAddress()` returns `io.netty.channel.unix.DomainSocketAddress`. See `HttpUdsTest` as an example. The same will happen for `ConnectionInfo` that we use for `network.transport` attribute in metrics and tracing. This is because internally we map ST type to Netty type before bootstrapping Netty (see `io.servicetalk.transport.netty.internal.BuilderUtils.toNettyAddress(Object)` but never map it back for `ConnectionInfo` or `ServerContext` types that return local/remote address.

How do you think, what should we do? Leave as-is and account for 2 types like here or do proper mapping to ST type in a follow-up PR?

Reply:
Yes, but the order doesn't matter. I can remove the 2nd check from these 2 files in that follow-up if we decide to do proper mapping.

You made a good point that if we change behavior, it may break existing users. What if they already look for Netty's `io.netty.channel.unix.DomainSocketAddress` type?

Reply:
Then they'll be broken. The question is whether this is something anybody will be doing? I suspect it's very rare but idk.

Reply:
I agree, UDS is a rare use-case. Ok, I will work on that change

Reply:
Made changes in #3309